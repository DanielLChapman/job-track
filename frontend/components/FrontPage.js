import React from 'react';

function FrontPage(props) {
    return (
        <div>
            Hi!
        </div>
    );
}

export default FrontPage;