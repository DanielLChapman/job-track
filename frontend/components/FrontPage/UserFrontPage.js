import React from 'react';
import CreateJob from '../Jobs/CreateJob';
import JobList from '../Jobs/JobList';
import SignOut from '../Signout';

function UserFrontPage({user, modalFill}) {
    return (
        <>
        
                    <section className="jobs-list">
                        <JobList jobs={user.jobs} user={user} modalFill={modalFill} />
                    </section></>
    );
}

export default UserFrontPage;