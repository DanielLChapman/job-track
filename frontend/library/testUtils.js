import casual from 'casual';

casual.seed(231313);

const fakeUser = (overrides) => ({
    __typename: 'User',
    id: '4234',
    name: casual.name,
    email: casual.email,
    ...overrides,
  });

  export {
    fakeUser
  }