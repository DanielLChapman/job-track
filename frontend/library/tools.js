export const convertNum = (num) => {
    let t = num.toString();
    if (t.length < 3) {
        t = "$" + t
        return t;
    }
    let result = [];
    let count = 1;
    for (let i = t.length - 1; i >= 0; i--) {
        result.unshift(t[i]);
        
        if (count % 3 === 0 && i !== 0) {
            result.unshift(',')
        }
        count++;
    }

    result.unshift('$')

    return result.join('');

}