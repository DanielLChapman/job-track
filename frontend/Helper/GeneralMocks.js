
import { CURRENT_USER_QUERY } from '../components/User';
import { fakeUser } from '../library/testUtils';

const fakeUserObj = fakeUser();
const notSignedInMocks = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: { data: { authenticatedItem: null } },
    },
  ];
  

  const signedInMocks = [
    {
      request: { query: CURRENT_USER_QUERY },
      result: { data: { authenticatedItem: fakeUser() } },
    },
  ];

  export {
    notSignedInMocks,
    signedInMocks,
}