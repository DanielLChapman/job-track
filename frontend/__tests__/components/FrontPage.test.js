import {act, screen, render} from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';

import {signedInMocks, notSignedInMocks} from '../../Helper/GeneralMocks'
import wait from 'waait';
import FrontPage from '../../components/FrontPage';
import { fakeUser } from '../../library/testUtils';

const user = fakeUser();

describe('It insures the front page renders text when not logged in', () => {
    it ('has a title, and two buttons when not logged in', () => {
        const { container, debug} = render(
            <MockedProvider >
                <FrontPage  />
            </MockedProvider>
        )

        expect(container).toMatchSnapshot();
    })

    it ('renders a different header when logged in', async () => {
        const {container: container2, debug: debug2} = render(
            <MockedProvider >
                <FrontPage />
            </MockedProvider>
        )
        const { container, debug} = render(
            <MockedProvider mocks={signedInMocks}>
                <FrontPage user={user} />
            </MockedProvider>
        )

        await act(async() => {
            await wait(0);
        });
        expect(container).toMatchSnapshot();
        expect(container).not.toBe(container2);
    })

    it ('makes sure that it is rendering the user front page when signed in', async () => {
        const { container, debug} = render(
            <MockedProvider mocks={signedInMocks}>
                <FrontPage user={user} />
            </MockedProvider>
        )

        await act(async() => {
            await wait(0);
        });

        const element = container.getElementsByClassName('jobs-list');

        expect(element.length).toBe(1);
    })
})