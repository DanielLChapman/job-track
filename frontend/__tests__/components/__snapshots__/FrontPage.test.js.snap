// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It insures the front page renders text when not logged in has a title, and two buttons when not logged in 1`] = `
<div>
  <section
    class="container"
  >
    <section
      class="header"
    >
      <h1>
        Job Track
      </h1>
      <div
        class="header-button-class"
      >
        <button
          class="sign-in-button"
          type="button"
        >
          Sign In / Up
        </button>
      </div>
    </section>
    <section
      class="info-section"
    >
      <section
        class="intro"
      >
        A website to keep track of your job applications!
      </section>
    </section>
  </section>
</div>
`;

exports[`It insures the front page renders text when not logged in renders a different header when logged in 1`] = `
<div>
  <section
    class="container"
  >
    <section
      class="header"
    >
      <h1>
        Job Track
      </h1>
      <div
        class="header-button-class"
      >
        <button
          class="sign-in-button"
          type="button"
        >
          Create Job
        </button>
        <button
          type="button"
        >
          Sign Out
        </button>
      </div>
    </section>
    <section
      class="jobs-list"
    >
      <ul
        class="job-view-container"
      />
    </section>
  </section>
</div>
`;
